### DISCUSSION TITLE
zh/datasets/pose/coco/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9965] [Docs]
# zh/datasets/pose/coco/

有关特殊 COCO-Pose 数据集的详细指南，请访问Ultralytics 。通过YOLO 了解其主要特征、结构以及在姿态估计任务中的使用。

https://docs.ultralytics.com/zh/datasets/pose/coco/

**Comment by WangDayeYa** [comment]
Hi, is the coco dataset that you train keypoint detection with directly from the coco website, or is there another dataset, is it necessary to have people present on the training images when performing keypoint detection, could you please help me with this question. Or does the coco dataset need to be filtered when performing keypoint detection in order to filter out images with people present? Hope to get your help on this.

**Reply by pderrenger** [reply]
Hello!

Yes, the COCO-Pose dataset used for keypoint detection with YOLOv8 is based directly on the COCO dataset available from the COCO website. Specifically, it leverages the COCO Keypoints 2017 images and labels to train models for pose estimation tasks.

For keypoint detection, it's indeed necessary for the images to have people present because the dataset supports 17 human keypoints to facilitate detailed pose estimation. There's no need to filter out images without people for this task, as the COCO-Pose dataset is already curated to include images annotated with keypoints relevant to pose estimation.

The Coco-Pose dataset does include images with people and has annotations for human keypoints. So, you don't need to filter out images without people. This dataset is specifically structured to aid in pose estimation tasks and includes a variety of images that are annotated for this purpose. Here’s a quick peek into how you can train a pose model with our dataset:

```python
from ultralytics import YOLO

# Load a pretrained pose estimation model
model = YOLO('yolov8n-pose.pt')

# Train the model on the COCO-Pose dataset
results = model.train(data='coco-pose.yaml', epochs=100, imgsz=640)
```

This command will use images that already have people in them and are labeled for pose estimation. If you need further details or assistance, feel free to ask. Happy modeling! 😊

**Reply by WangDayeYa** [reply]
Hello, could you please provide the download link of the coco-pose dataset, so that I can start to perform a replay of the key point detection? How did you come up with those evaluation indicators? What are the specific data set requirements and parameter configurations?

**Reply by pderrenger** [reply]
@WangDayeYa hey there! 🌟

For the COCO-Pose dataset, you can start by downloading the COCO 2017 images from the official [COCO Dataset](https://cocodataset.org/#download) website and then use our specific annotations for pose estimation tasks. Here's a snippet to help you with the download process:

```python
from ultralytics.utils.downloads import download
from pathlib import Path

# Dataset root directory
dir = Path('../datasets/coco-pose')

# URLs for downloading COCO 2017 labels and images
urls = ['https://github.com/ultralytics/yolov5/releases/download/v1.0/coco2017labels-pose.zip',  # Labels
        'http://images.cocodataset.org/zips/train2017.zip',  # 118k training images
        'http://images.cocodataset.org/zips/val2017.zip',  # 5k validation images
        'http://images.cocodataset.org/zips/test2017.zip']  # 41k test images (optional)

# Download process
download(urls, dir=dir / 'images', threads=3)
```

The evaluation indicators, such as Object Keypoint Similarity (OKS), are standardized metrics provided by COCO to facilitate performance comparison across different models on pose estimation tasks.

For training on the COCO-Pose dataset, your dataset YAML file (`coco-pose.yaml`) is crucial. Here's how you'd train a YOLOv8n-pose model for 100 epochs with images sized at 640:

```bash
yolo detect train data=coco-pose.yaml model=yolov8n-pose.pt epochs=100 imgsz=640
```

This setup assumes `coco-pose.yaml` is properly pointing to your COCO-Pose dataset's directory and the dataset files. The YAML configuration should match the one you found in our documentation, customized to your specific paths if necessary.

Hope this gets you started on the right foot with keypoint detection! If you've got more questions, feel free to reach out. Happy detecting! 😊

**Reply by WangDayeYa** [reply]
Okay, According to your description, are you guys using the corresponding labeling information of train2017 and val2017 in the labels of coco2017labels-pose.zip in yolov5 to filter all the coco dataset images in order to get the coco-pose dataset for training.In other words, are you training and validating the dataset images corresponding to the labels file of coco2017labels-pose.zip in yolov5? may I ask how many rounds of training on the coco-pose dataset did you get your data results inside the document?Thank you.

**Reply by pderrenger** [reply]
@WangDayeYa hey there! 👋

Yes, you're correct! We use the labeling information from `train2017` and `val2017` found in the `coco2017labels-pose.zip` for filtering the COCO dataset images to obtain the COCO-Pose dataset. This enables our models, such as YOLOv8n-pose, to be trained and validated specifically for pose estimation tasks using these annotated images.

Regarding the number of training epochs, the performance results documented are typically achieved after comprehensive training, often spanning hundreds of epochs to maximize accuracy and model effervescence. The specific number of epochs may vary depending on the model and training configurations.

For example, to train a pose estimation model like YOLOv8n-pose, you might use:
```bash
yolo detect train data=coco-pose.yaml model=yolov8n-pose.pt epochs=100 imgsz=640
```
This command initiates training with the COCO-Pose dataset, targeting 100 epochs.

Each model listed, from YOLOv8n-pose through YOLOv8x-pose-p6, is rigorously trained to ensure optimal performance, with detailed evaluations conducted to generate the documentation results. For up-to-date performance metrics, it's a good idea to refer to our official documentation and experiment logs. 😊

Thanks for your question, and best of luck with your pose estimation projects!

**Comment by xcf1** [comment]
I used my pre-trained model to infer the test-dev2017 dataset and uploaded the inference results to the COCO official website for evaluation, but I encountered an error stating “Results do not correspond to current coco set.” However, my JSON file format complies with the requirements of the COCO official website. Can you help me solve this issue?

**Reply by glenn-jocher** [reply]
@xcf1 ensure your results JSON file uses image IDs from the COCO test-dev2017 set (20,288 images) and follows the exact keypoints format required. The error often occurs when using validation set IDs or incorrect keypoint ordering. Double-check against the COCO Keypoints Evaluation page submission requirements.
